quarkus-kubernetes extension throws NullPointerException during maven test phase

**Describe the bug**
When compiling a very basic Quarkus application (see reproduce steps), running "mvn package" throws a NullPointerException in the test phase:

```
Build failure: Build failed due to errors
        [error]: Build step io.quarkus.kubernetes.deployment.KubernetesProcessor#build threw an exception: java.lang.NullPointerException
Caused by: java.lang.NullPointerException
```

**To Reproduce**
Steps to reproduce the behavior:
1. Create a sample application using `mvn quarkus:create`
2. Run `mvn quarkus:add-extension -Dextensions="quarkus-kubernetes"`
3. Run `mvn compile package`

**Environment (please complete the following information):**
 - Output of `java -version`: 1.8.0_222
 - GraalVM version (if different from Java):  19.1.1
 - Quarkus version or git rev: 0.20.0

**Additional context**
This problem does not exist in Quarkus 0.19.1

--- a/core/deployment/src/main/java/io/quarkus/deployment/IsNormal.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/IsNormal.java
@@ -1,0 +1,24 @@
+package io.quarkus.deployment;
+
+import java.util.function.BooleanSupplier;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.runtime.LaunchMode;
+
+/**
+ * boolean supplier that returns true if the application is running in normal
+ * mode. Intended for use with {@link BuildStep#onlyIf()}
+ */
+public class IsNormal implements BooleanSupplier {
+
+    private final LaunchMode launchMode;
+
+    public IsNormal(LaunchMode launchMode) {
+        this.launchMode = launchMode;
+    }
+
+    @Override
+    public boolean getAsBoolean() {
+        return launchMode == LaunchMode.NORMAL;
+    }
+}

--- a/extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java
+++ b/extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java
@@ -20,6 +20,7 @@ import io.dekorate.kubernetes.annotation.KubernetesApplication;
 import io.dekorate.kubernetes.generator.DefaultKubernetesApplicationGenerator;
 import io.dekorate.processor.SimpleFileWriter;
 import io.dekorate.project.Project;
+import io.quarkus.deployment.IsNormal;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.ApplicationInfoBuildItem;
@@ -37,7 +38,7 @@ class KubernetesProcessor {
     @Inject
     BuildProducer<FeatureBuildItem> featureProducer;
 
-    @BuildStep
+    @BuildStep(onlyIf = IsNormal.class)
     public void build(ApplicationInfoBuildItem applicationInfo,
             KubernetesConfig kubernetesConfig,
             List<KubernetesPortBuildItem> kubernetesPortBuildItems,

